nuxeo-runtime/nuxeo-runtime/src/main/java/org/nuxeo/runtime/AbstractRuntimeService.java

```diff
@@ -158,83 +175,148 @@ public synchronized void start() {
+                   if (event.id == RuntimeServiceEvent.RUNTIME_ABOUT_TO_START) {
+                       onStart();
+                   } else if (event.id == RuntimeServiceEvent.RUNTIME_ABOUT_TO_RESUME) {
+                       onResume();
+                   } else if (event.id == RuntimeServiceEvent.RUNTIME_ABOUT_TO_STANDBY) {
+                       onStandby();
+                   } else if (event.id == RuntimeServiceEvent.RUNTIME_STOPPED) {
+                       Framework.removeListener(this);
+                   }
+                }
+
+                void onStart() {
+                    wdog = Log4JHelper.configureAndWatch(delay);
+                    if (wdog == null) {
+                        log.warn("Failed to configure log4j.xml change detection");
+                    } else {
+                        log.info("Configured log4j.xml change detection with a delay of " + delay + "s");
+                    }
+                }
+
+                void onResume() {
+                    if (wdog != null) {
+                        return;
+                    }
+                    wdog = Log4JHelper.configureAndWatch(delay);
+                }
+
+                void onStandby() {
+                    if (wdog == null) {
+                        return;
+                    }
+                    try {
+                        wdog.cancel();
+                    } finally {
+                        wdog = null;
+                    }
+                }
+            });
+
         }
         log.info("Starting Nuxeo Runtime service " + getName() + "; version: " + getVersion());
 
         Framework.sendEvent(new RuntimeServiceEvent(RuntimeServiceEvent.RUNTIME_ABOUT_TO_START, this));
-        ServicePassivator.passivate()
-                         .withQuietDelay(Duration.ofSeconds(0))
-                         .monitor()
-                         .withTimeout(Duration.ofSeconds(0))
-                         .withEnforceMode(false)
-                         .await()
-                         .proceed(() -> {
-                             try {
-                                 doStart();
-                                 startExtensions();
-                             } finally {
-                                 Framework.sendEvent(
-                                         new RuntimeServiceEvent(RuntimeServiceEvent.RUNTIME_STARTED, this));
-                                 isStarted = true;
-                             }
-                         });
+        try {
+            state = State.LOADING;
+            doStart();
+            startExtensions();
+        } finally {
+            state = State.STANDBY;
+            Framework.sendEvent(new RuntimeServiceEvent(RuntimeServiceEvent.RUNTIME_STARTED, this));
+        }
     }
 
     @Override
-    public synchronized void stop() {
-        if (!isStarted) {
+    public void stop() {
+        if (state.ordinal() < State.STANDBY.ordinal()) {
```

Comment:
If order matters then please add a comment on the `State` enum about it. I think I'd prefer an explicit list of states instead though.